clidocstool_md.go

```diff
@@ -193,19 +193,30 @@ func mdCmdOutput(cmd *cobra.Command, old string) (string, error) {
 				name = mdMakeLink(name, f.Name, f, isLink)
 				fmt.Fprintf(b, "%s, ", name)
 			}
-			name := "`--" + f.Name
-			if f.Value.Type() != "bool" {
-				name += " " + f.Value.Type()
-			}
-			name += "`"
+			name := "`--" + f.Name + "`"
 			name = mdMakeLink(name, f.Name, f, isLink)
+
+			var ftype string
+			switch f.Value.Type() {
+			case "bool":
+			case "stringArray":
+				ftype = "list"
+			case "stringSlice":
+				ftype = "list/strings"
```

Comment:
We should review our uses of `stringArray` vs `stringSlice`. I think (due to oversight when adding flags) we haven't always checked which one we _wanted_ to use, and there's some differences.

If I'm not mistaken, 

- both can be used to pass multiple values by setting the same flag multiple times (`--opt foo --opt bar` means `[]string{"foo", "bar"}`)
- one of them _also_ accepts comma-separated lists (`--opt foo,bar --opt baz` means `[]string{"foo", "bar", "bar"}`)

And then, we have `string` values that silently overwrite prior values if the flag is passed multiple times (`--opt foo --opt bar` means `"bar"`)

I'm slightly wondering if we're always able to use the generic names here (could there be cases where `string` or `list` cannot be used?)

Reply:
My changes should not be there because the type will only be shown in docs and not in the CLI:

```text
      --build-arg stringArray       Set build-time variables
```

So should be done upstream in docker/cli I guess in [opts](https://github.com/docker/cli/blob/master/opts/opts.go).

> We should review our uses of `stringArray` vs `stringSlice`. I think (due to oversight when adding flags) we haven't always checked which one we _wanted_ to use, and there's some differences.

We use `stringArray` in buildx but on docker/cli we use `opts.ListOpts`: https://github.com/docker/cli/blob/c780f7c4abaf67034ecfaa0611e03695cf9e4a3e/cli/command/image/build.go#L42 which has a `list` type: https://github.com/docker/cli/blob/c780f7c4abaf67034ecfaa0611e03695cf9e4a3e/opts/opts.go#L112-L115. I wonder if we should do the same on buildx repo or at least have a specific and understandable override for `Type`  to have `list` instead of `stringArray`.

> If I'm not mistaken,
> 
>     * both can be used to pass multiple values by setting the same flag multiple times (`--opt foo --opt bar` means `[]string{"foo", "bar"}`)
> 
>     * one of them _also_ accepts comma-separated lists (`--opt foo,bar --opt baz` means `[]string{"foo", "bar", "bar"}`)

Yes that's it: https://github.com/docker/buildx/issues/922#issuecomment-1016260687

Reply:
> My changes should not be there because the type will only be shown in docs and not in the CLI:

Ah, yes, of course. That would be inconsistent